src/Activity.js

```diff
@@ -1,54 +1,126 @@
-  //   return parseFloat(((userStepsByDate.numSteps * userRepo.strideLength) / 5280).toFixed(1)); //how is userRepo.strideLength even being accessed?
-  //   // userRepo is array of user objects but how does that have a strideLength property?
-  //   // userRepo is improperly named
-  // }
+  //// begin unused methods
+  getMilesByDate(id, date, user, userRepo) {  //returns miles walked on a specific date
+    // THIS IS NEVER USED needs to be displayed on page
+    let userStepsByDate = this.getUserDataForDay(id, date, 'numSteps').amount;
+    return parseFloat(((userStepsByDate * user.strideLength) / 5280).toFixed(1));
+  }
   // // not used
+
+  // replaced
   // getActiveMinutesByDate(id, date) { //return total minues active on a date for a specific user
   //   let userActivityByDate = this.activityData.find(data => id === data.userID && date === data.date);
   //   return userActivityByDate.minutesActive;
   // }
-  // calculateActiveAverageForWeek(id, date, userRepo) { //return average active minutes for all users
-  //   return parseFloat((userRepo.getWeekFromDate(date, id, this.activityData).reduce((acc, elem) => { //acc and elem are terrible names
-  //     return acc += elem.minutesActive;
-  //   }, 0) / 7).toFixed(1));
-  // }
-  // accomplishStepGoal(id, date, userRepo) { // returns true if user steps = daily step goal (should be >=)
-  //   let userStepsByDate = this.activityData.find(data => id === data.userID && date === data.date);
-  //   if (userStepsByDate.numSteps === userRepo.dailyStepGoal) { //>= not ===
-  //     return true;
-  //   }
-  //   return false
-  // }
-  // getDaysGoalExceeded(id, userRepo) { // returns array of dates user exceeding goal steps
-  //   return this.activityData.filter(data => id === data.userID && data.numSteps > userRepo.dailyStepGoal).map(data => data.date);
-  // }
-  // getStairRecord(id) { // returns highest number of stairs user as ever done on single day
-  //   return this.activityData.filter(data => id === data.userID).reduce((acc, elem) => (elem.flightsOfStairs > acc) ? elem.flightsOfStairs : acc, 0);
-  // }
-  //// end not needed methods
+
+
+  calculateActiveAverageForWeek(id, date, userRepo) { //return average active minutes for all users
+    // console.log(userRepo.getWeekFromDate(date, id, this.activityData))
+    // console.log(this.getUserWeekData(id, date, 'minutesActive'))
+    //
+    const weekData = this.getUserWeekData(id, date, 'minutesActive');
+    const weeklyAverage = (weekData.reduce((sum, day) => {
+      return sum += day.amount
+    }, 0) / 7)
+    return parseFloat(weeklyAverage.toFixed(1));
+    // return `{parseFloat(weeklyAverage.toFixed(1))} minutes active`
+    // return parseFloat((userRepo.getWeekFromDate(date, id, this.activityData).reduce((acc, elem) => { //acc and elem are terrible names
+    //   return acc += elem.minutesActive;
+    // }, 0) / 7).toFixed(1));
+  }
```

Comment:
This is for just one user right? I like the logic for it a lot and might use it in Sleep.

Reply:
Yeah, get logic here!